app/modules/firebase-db.js

```diff
@@ -54,20 +56,45 @@ export function saveProfileToFb(data) {
       .catch((err) => {
         throw new Error(`Error getting cat profile: ${err}`);
       })
-    );
+  );
 }
 
-export function fetchCatsFromFb(uid) {
+function merge(cats, snaps) {
+  return _.map(cats, (cat, index) => _.assign(
+    {},
+    cat,
+    snaps[index] || {}
+  ));
```

Comment:
Either way, that biz deserves a test.

On Sun, Aug 28, 2016 at 21:57 Abhi Gulati <notifications@github.com> wrote:

> In app/modules/firebase-db.js
> <https://github.com/chrisbodhi/mewment/pull/48#discussion_r76553271>:
>
> >  }
> >
> > -export function fetchCatsFromFb(uid) {
> > +function merge(cats, snaps) {
> > +  return _.map(cats, (cat, index) => _.assign(
> > +    {},
> > +    cat,
> > +    snaps[index] || {}
> > +  ));
>
> I think with object rest/spread, the right side of the arrow fn could just
> be
>
> {
>   ...cat,
>   ...snaps[index]
> }
>
> Just an FYI. I wouldn't advocate that change here; the explicit fallback
> to empty object is probably more clear.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/chrisbodhi/mewment/pull/48/files/0677ab3d284f6a9e281b68f6d02cbe7b36b3284c#r76553271>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD4hERosT2QCOHN7ai_qBae3PXx25GFSks5qkmbFgaJpZM4JvILi>
> .
>